src/containers/PipelinesDropdown/PipelinesDropdown.test.js

```diff
@@ -0,0 +1,254 @@
+/*
+Copyright 2019 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import { fireEvent, render } from 'react-testing-library';
+import { Provider } from 'react-redux';
+import configureStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import PipelinesDropdown from './PipelinesDropdown';
+import * as API from '../../api';
+
+const props = {
+  id: 'pipelines-dropdown'
+};
+
+const pipelinesByNamespace = {
+  blue: {
+    'pipeline-1': 'id-pipeline-1',
+    'pipeline-2': 'id-pipeline-2'
+  },
+  green: {
+    'pipeline-3': 'id-pipeline-3'
+  }
+};
+
+const pipelinesById = {
+  'id-pipeline-1': {
+    metadata: {
+      name: 'pipeline-1',
+      namespace: 'blue',
+      uid: 'id-pipeline-1'
+    }
+  },
+  'id-pipeline-2': {
+    metadata: {
+      name: 'pipeline-2',
+      namespace: 'blue',
+      uid: 'id-pipeline-2'
+    }
+  },
+  'id-pipeline-3': {
+    metadata: {
+      name: 'pipeline-3',
+      namespace: 'green',
+      uid: 'id-pipeline-3'
+    }
+  }
+};
+
+const pipelinesStoreDefault = {
+  pipelines: {
+    byId: pipelinesById,
+    byNamespace: pipelinesByNamespace,
+    isFetching: false
+  }
+};
+
+const pipelinesStoreFetching = {
+  pipelines: {
+    byId: pipelinesById,
+    byNamespace: pipelinesByNamespace,
+    isFetching: true
+  }
+};
+
+const namespacesByName = {
+  blue: '',
+  green: ''
+};
+
+const namespacesStoreBlue = {
+  namespaces: {
+    byName: namespacesByName,
+    selected: 'blue'
+  }
+};
+
+const namespacesStoreGreen = {
+  namespaces: {
+    byName: namespacesByName,
+    selected: 'green'
+  }
+};
+
+const initialTextRegExp = new RegExp('select pipeline', 'i');
+
+const middleware = [thunk];
+const mockStore = configureStore(middleware);
+
+beforeEach(() => {
+  jest.spyOn(API, 'getPipelines').mockImplementation(() => pipelinesById);
+});
+
+it('PipelinesDropdown renders items based on Redux state', () => {
+  const store = mockStore({
+    ...pipelinesStoreDefault,
+    ...namespacesStoreBlue
+  });
+  const { getByText, queryByText } = render(
+    <Provider store={store}>
+      <PipelinesDropdown {...props} />
+    </Provider>
+  );
+  // View items
+  fireEvent.click(getByText(initialTextRegExp));
+  Object.keys(pipelinesByNamespace.blue).forEach(item => {
+    expect(queryByText(new RegExp(item, 'i'))).toBeTruthy();
+  });
+  // Select item
+  fireEvent.click(getByText(/pipeline-1/i));
+  Object.keys(pipelinesByNamespace.blue).forEach(item => {
```

Comment:
It's not clear to me exactly what this is testing, can you explain?

Reply:
@AlanGreene I wanted this to test the item selection process. So, a user (1) clicks the dropdown to view the items (2) clicks an item to select it, then (3) the item is displayed in the dropdown and all of the other items disappear.

This section is testing part (3) that all of the other items disappear, and the dropdown displays only the selected item.

Is this overkill for UI unit tests? I'm a little new to them, so I try to air on the cautiously thorough side when testing.

Reply:
Ah I see. Yes no need for that here, we can assume the underlying dropdown component works as expected so all we're really concerned with for these tests are the pipeline-specific pieces.

Reply:
Ok. I'll make this update to the `ServiceAccountsDropdown` and the `TooltipDropdown` tests as well